### ISSUE TITLE
[homekit] OH3 homekit Colortemperature and hue issues

### ISSUE THREAD

**Issue by pug304** [openhab/openhab-addons#9799]
on a oh3.1 snapshot Build #2135 i've installed two lamps: xiaomi mi desk lamp 1s and one Philips Hue. Both of them are able to adjust colortemperature and brightness, the hue is able to change color/hue additional. Both lamps are working well via OH Pages (switch on/brightness,/temperature/hue).

When integrating in Homekit all is working fine with setting metadata to tags lighting, lighting.OnState and lighting.brightness. Once you add tags lighting.colortemperature or lighting.hue, checking status in Homekit status of icons hang at refresh (Aktualisieren on german), if you press buttons of the items Homekit app close immediate. Restart of homekit app will close the app immediate again. Remove the mentioned tags everything works like before.

Addons are standard (Xiaomi mi and Hue) coming with OH3. Things/Items/Model are defined via UI.

log with log:set TRACE io.github.hapjava/log:tail io.github.hapjava gives tons of log info, it seems communication is working. I've snip the log and have save it into a local file. If it ie helpful i will share.

Not an important bug due to Homekit is more or less a toy ;) 

edit: metatag coltemp didn't work before with other snapshot versions. i didn't take care about this, with the second lamp and the same issue occured i think it is reasonable now to have a look at it

**Comment by yfre** [comment]
probably related https://community.openhab.org/t/homekit-color-temperature-with-hue-bulbs/105658/11?u=yfre

**Comment by pug304** [comment]
> 
> 
> probably related https://community.openhab.org/t/homekit-color-temperature-with-hue-bulbs/105658/11?u=yfre

yes Eugen, thats absulte the same issue! Looking forward to your investigation!

**Comment by yfre** [comment]
@pug304  
issue identified (https://community.openhab.org/t/homekit-color-temperature-with-hue-bulbs/105658/16?u=yfre), fix for underlaying library is submitted, but it will take some time until new version of that lib gets released and we can switch to it in openHAB.

**Comment by pug304** [comment]
> 
> 
> @pug304
> issue identified (https://community.openhab.org/t/homekit-color-temperature-with-hue-bulbs/105658/16?u=yfre), fix for underlaying library is submitted, but it will take some time until new version of that lib gets released and we can switch to it in openHAB.

sounds good!

**Comment by Sirmike2** [comment]
The issue still exists. I have tested with the snapshot 3.2.0 S2604 after change to official Java HAP lib release.
Also it is strange, if I create a point that is named Colortemp and I add the Metadata for HomeKit (light.colortemperature) this point will show the correct number for the HomeKit setting, but if make the same Point with a different name it will not work and I get also the not responding.

**Comment by yfre** [comment]
> Also it is strange, if I create a point that is named Colortemp and I add the Metadata for HomeKit (light.colortemperature) this point will show the correct number for the HomeKit setting, but if make the same Point with a different name it will not work and I get also the not responding.

this is strange. there is not link whatsoever to the name of the item in the code. can you try 2-3 different names, ideally totally new and never used before?

and do understand correctly, that for Colortemp is works as expected and you see it in home app?

**Comment by Sirmike2** [comment]
I have tried a lot of names, always the same. Only the name Colortemp is working. This Point is not linked to some channel. It was only for me to check what values I will get from HomeKit app and this works with the name Colortemp. I can select the colortemperature in the App and the Point shows the values (72 - 400).

**Comment by yfre** [comment]
can you share screenshots of now working point? i will try to reproduce on my side

**Comment by Sirmike2** [comment]
Two screenshots one with one named Testpoint and one with Colortemp.
![Testpoint](https://user-images.githubusercontent.com/30008506/144621755-f680cb73-47be-4350-ba7c-b459abfec9c7.jpg)
![Colortemp](https://user-images.githubusercontent.com/30008506/144621772-ecb397ab-124a-4576-b160-992399661d2d.jpg)

**Comment by Sirmike2** [comment]
Some more:
![Testpoint_2](https://user-images.githubusercontent.com/30008506/144622285-2f7957bb-3f14-4775-8bcb-ac34af7858f5.jpg)
![Colortemp_2](https://user-images.githubusercontent.com/30008506/144622317-44125606-ab95-4ec8-a32c-5eebf64550be.jpg)

If you need more, thats no problem !

**Comment by yfre** [comment]
very helpful, we are getting closer. but i have 2 more requests:
- can you try to put a value to Testpoint before you add homekit. so, that instead of "NULL" you have a number there
- can you show me the the complete group "TestlampWSIKEAofSweden"

**Comment by Sirmike2** [comment]
I did it and it will be now also ok for Testpoint. So if in some Point for colortemperature is NULL than it will break. 
![TestlampeWS](https://user-images.githubusercontent.com/30008506/144637052-121f6e22-56c1-4703-ade4-9170dc185ea2.jpg)
But if I link this to the deCONZ/HUE channel for Colortemperture it will not work. This is probably because it needs kelvin and not mired. So I have to look for a script or rule that convert it.

**Comment by Sirmike2** [comment]
Thanks for help last time. It will work now with script to change from mired to kelvin. I have one more question. I have seen, that in ColorTemperatureCharacteristic.java the defaults for colortemp range are 50 and 400. Is it possible to change this values because I have red something at AccessoryWithColorTemperature.java. I have always worked with Homebridge and FHEM. There is the possibility to change with the mapping the Values, that HomeKit App shows only this values with the cycle. The most Lamps have Colortemp from lets say ~2000 to ~7000 K or ~500 to ~140.

**Comment by yfre** [comment]
@Sirmike2  right, the configuration option of min/max value was missing. just added it and committed the change. i also will set the value to minValue is openHAB item is NULL. should reduce issues with new items with no value at the beginning

**Comment by hpop** [comment]
> It will work now with script to change from mired to kelvin.
@Sirmike2  Did you find a solution to convert the values? I was looking into it and didn't find a solution.

**Comment by Sirmike2** [comment]
In principial I created a second point for my lamp, for example lamp_colortemp_homekit. My first point name is lamp_colortemp. This one is linked to the colortemp channel of the real lamp. So what I did, was only a rule, if I change my colortemp at lamp_colortemp (Kelvin) it will convert to lamp_colortemp_homekit (Mired). 1 Kelvin = 1000000 / Mired
For example :
`val kelvin = (1000000 / mired_homekit).intValue`
And of course, if homekit changes my point lamp_colortemp_homekit, it will convert my first point lamp_colortemp
